components/UI/Drawer/Drawer.tsx

```diff
@@ -125,6 +125,22 @@ const Drawer = () => {
       geometry.location.lng,
     ]).format();
 
+    const openSourceUrl = () => {
+      if (source.channel === "twitter" && source.extra_parameters) {
+        const extraParams = JSON.parse(source.extra_parameters);
+        if (extraParams.user_id && extraParams.tweet_id) {
+          window.open(
```

Comment:
Did you try to make this routing with next/router? window.open may cause problems.

you can use useRouter() hook.